server.js

```diff
@@ -30,7 +30,61 @@ server.get('/api/users/:id', (req, res) => {
     .catch(() => {
       res.status(500).json({ error: "The user information could not be retrieved." })
     })
-})
+});
+
+server.post('/api/users', (req, res) => {
+  const { name } = req.body;
+  if ( { name } === null ) {
+    res.status(400).json({ errorMessage: "Please provide name for the user." })
+  } else {
+      users.insert({ name })
+      .then(response => {
+        res.status(201).json(response)
+      })
+      .catch(() => {
+        res.status(500).json({ error: "There was an error while saving the user to the database." })
+      })
+  }
+});
+
+server.delete('/api/users/:id', (req, res) => {
+  const { id } = req.params;
+  users.remove(id)
+    .then(response => {
+      if (response > 0) {
+        res.status(200).json(response)
+      } else {
+        res.status(404).json({ message: "The user with the specified ID does not exist." })
+      }
+    })
+    .catch(() => {
+      res.status(500).json({ error: "The user could not be removed."})
+    })
+});
+
+server.put('/api/users/:id', (req, res) => {
+  const { id } = req.params;
+  const { name } = req.body;
+  if ( { name } === null ) {
```

Comment:
Another way to write this would be simple if (!name) and that would be sufficient.